GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+            }
+            else
+            {
+                Console.WriteLine("User not found");
+            }
+
+            #endregion
+        }
+
+        private static async Task PrintUsersManager( User user)
+        {
+            #region Get user's manager and direct reports, group memberships and role memberships
+
+            // ***********************************************************************
+            // NOTE:  This requires User.Read.All permission scope, or Directory.Read.All or Directory.AccessAsUser.All
+            // Group membership requires Group.Read.All or Directory.Read.All (the latter is required for role memberships)
+            // Code snippet also demonstrates paging through user's direct reports
+            // ***********************************************************************
+
+            // manager and reports...
+            try
+            {
+                Console.WriteLine("\nRetrieving signed in user's Manager and Direct Reports");
+                IUserFetcher userFetcher = user as IUserFetcher;
+                IDirectoryObject manager = await userFetcher.Manager.ExecuteAsync();
+                IPagedCollection<IDirectoryObject> reports = await userFetcher.DirectReports.ExecuteAsync();
+
+                if (manager is User)
+                {
+                    Console.WriteLine("\n  Manager (user):" + ((IUser) (manager)).DisplayName);
+                }
+                else if (manager is Contact)
+                {
+                    Console.WriteLine("\n  Manager (contact):" + ((IContact) (manager)).DisplayName);
+                }
+                else
+                {
+                    Console.WriteLine("\n  User has no manager :)");
+                }
+
+                if (reports != null)
+                {
+                    Console.WriteLine("\n  Direct reports:");
+                }
+                do
+                {
+                    List<IDirectoryObject> directoryObjects = reports.CurrentPage.ToList();
+                    foreach (IDirectoryObject directoryObject in directoryObjects)
+                    {
+                        if (directoryObject is User)
+                        {
+                            Console.WriteLine("\n    " + ((IUser) (manager)).DisplayName);
+                        }
+                        else if (directoryObject is Contact)
+                        {
+                            Console.WriteLine("\n    " + ((IContact) (manager)).DisplayName);
+                        }
+
+                    }
+                    reports = await reports.GetNextPageAsync();
+                } while (reports != null);
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting user's manager and reports {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            // group and role memberships
```

Comment:
done

---
In reply to: [80542136](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r80542136) [](ancestors = 80542136)